tracking/translations_parser/parser.py

```diff
@@ -0,0 +1,288 @@
+                desc.append(text)
+            self.description = " ".join(desc)
+
+            # Try to parse all following config lines as YAML
+            logger.debug("Reading Marian configuration.")
+            config_yaml = ""
+            while ("config",) in headers:
+                if "Model is being created" in text:
+                    headers, text = next(logs_iter)
+                    break
+                config_yaml += f"{text}\n"
+                headers, text = next(logs_iter)
+            try:
+                self.config = yaml.safe_load(config_yaml)
+            except Exception as e:
+                raise Exception(f"Invalid config section: {e}")
+
+        for publisher in self.publishers:
+            publisher.open(self)
+
+        # Publish optional metrics
+        if self.metrics:
+            for publisher in self.publishers:
+                publisher.handle_metrics(self.metrics)
+
+        # Iterate until the end of file to find training or validation logs
+        while True:
+            try:
+                try:
+                    training = self.parse_training_log(headers, text)
+                    if not training:
+                        self.parse_validation_log(headers, text)
+                except ValueError as e:
+                    logger.warning(f"Line {self._current_index} could not be stored: {e}.")
+                finally:
+                    headers, text = next(logs_iter)
+            except StopIteration:
+                break
+
+        # Report incomplete validation logs
+        if self._validation_entries.keys():
+            logger.warning(
+                "Some validation data is incomplete with the following epoch/up couples:"
+            )
+            for epoch, up in self._validation_entries.keys():
+                logger.warning(f"* Ep. {epoch}, Up. {up}")
+
+        self.parsed = True
+
+    @property
+    def logs_str(self):
+        return "\n".join(
+            "".join(f"[{key}] {val}\n" for val in values)
+            for key, values in self.indexed_logs.items()
+        )
+
+    @property
+    def output(self):
+        if not self.parsed:
+            raise Exception("Please run the parser before reading the output")
+        return TrainingLog(
+            run_date=self.run_date,
+            configuration=self.config,
+            training=self.training,
+            validation=list(self.validation),
+            logs=self.indexed_logs,
+        )
+
+    def run(self):
+        """
+        Parse the log lines.
+        """
+        try:
+            self._parse()
+        except StopIteration:
+            # A StopIteration can be raised if some required lines are never found.
+            raise ValueError("Logs file ended up unexpectedly")
+
+        count = sum(len(vals) for vals in self.indexed_logs.values())
+        logger.info(f"Successfully parsed {count} lines")
+        logger.info(f"Found {len(self.training)} training entries")
```

Comment:
Thanks for reporting these stats! It might be valuable to also report how many lines failed to publish (I think the only case when this happen is an error in W&B).